include/alpaka/acc/Tag.hpp

```diff
@@ -5,12 +5,18 @@
 #pragma once
 
 #include "alpaka/core/BoostPredef.hpp"
+#include "alpaka/core/Concepts.hpp"
 
 #include <iostream>
 #include <type_traits>
 
+namespace alpaka
+{
+    struct ConceptTag;
+} // namespace alpaka
+
 #define CREATE_ACC_TAG(tag_name)                                                                                      \
-    struct tag_name                                                                                                   \
+    struct tag_name : public concepts::Implements<alpaka::ConceptTag, tag_name>                                       \
```

Comment:
I need to introduce the `ConceptTag` to make the class unique. Otherwise the only property to identify a alpaka tag is the helper static class function `get_name`, which is not unique.

Maybe we should think about to rename `alpaka::concepts` because it confuses now, than we have C++20 concepts.

Reply:
In an offline discussion with @fwyzard he suggested `alpaka::interfaces`. I like the idea, because actual this mechanism imitates traits from other languages but traits has already a meaning in alpaka. Therefore interface is a good solution because it sounds like java interfaces which is pretty similar.